### ISSUE TITLE
New Use Case:Ocean fronts demonstrating modified Hausdorff

### ISSUE THREAD

**Issue by j-opatz** [dtcenter/METplus#1494]
## Describe the New Use Case ##
This is the final use case that was being tracked by the umbrella issue #800 to bring marine verification into the METplus suite. 

This use case will replicate the verification work done for [ocean fronts](https://polar.ncep.noaa.gov/global/fronts/). While this largely relies on plotting output and will be completed at a later time in METplotpy, the focus will remain largely on figuring out how to replicate the modified Hausdorff.

### Use Case Name and Category ###
Use case name will be determined once input sources are identified.

This use case will reside in the marine_and_cryosphere category

### Input Data ###
*List input data types and sources.*
*Provide a total input file size, keeping necessary data to a minimum.* 

### Acceptance Testing ###
This use case will be successful if it replicates the modified Hausdorff values with the same input data as the website.
*As use case develops, provide a run time here*

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*

### Sub-Issues ###
Consider breaking the new feature down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**
- [ ] Select **privacy**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## New Use Case Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept your changes. Merge branch into **develop**.
- [ ] Create a second pull request to merge **develop** into **develop-ref**, following the same steps for the first pull request.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by georgemccabe** [comment]
@j-opatz , @mrinalbiswas, has this work been completed?

**Comment by mrinalbiswas** [comment]
@georgemccabe No, this work is not completed. I will let @j-opatz comment on the limitations of MET that is needed for the use case.

**Comment by georgemccabe** [comment]
Thanks @mrinalbiswas . @j-opatz , I noticed this issue has the "priority:high" and "required: FOR OFFICIAL RELEASE" labels but is not assigned to a release milestone. Could you adjust the labels/milestones as needed?

**Comment by j-opatz** [comment]
In an email with EMC on March 20th, a proposed path forward was discussed. Below is a summary of the idea, along with the constraints:

Now that Python is supported on EVS, we can pursue a use case that utilizes UserScript, essentially allowing the Python script to do the heavy lifting while being kicked off within the METplus environment.

With that route comes some constraints: only the Python packages listed in the March 20th email can be utilized. If anything appears outside the provided list, we need to a) import the critical package logic to the Python script and eliminate the outside reliance b) try and get the package added to the EVSv2 approved package list (but it's a little late for this route) or c) resume progress later when more packages are allowed, MET can perform point-to-point calculations, etc. 

From a quick check of the existing Hausdorff script, I see that one package in the script (`seawater`) is not present on the EVS-allowed Python packages. But the only method that package provides is a temperature from potential temperature conversion. More research needs to be done to see how much would have to be pulled from `seawater` (looks like at least a subroutine calculation of potential temperature [ptmp] and adiabatic temperature gradient [adtg]), but initial investigations look good.

**Comment by j-opatz** [comment]
After a brief review of this use case and its dependencies, it looks like this could be a candidate for UserScript usage in METplus. There are a few dependencies, listed here:

- numpy
- glob
- datetime
- pandas
- matplotlib
- netCDF4
- scipy
- os
- sys
- warnings

Some of these, such as `warnings` and `os`, can probably be eliminated through alternate scripting. 

All of the dependencies listed fall within the Python packages supported by NCO, which is great news! There is one, however, that is not; `seawater`, as listed in my last update, would need to have all relevant functionality transferred to the script. While this would create a bit of work, it can still be accomplished, as `seawater` is [opensource and available through Github](https://github.com/pyoceans/python-seawater/tree/main).

The following are all routines in the `seawater` that would need to be copied:

- ptmp (calculates potential temperature)
- adtg (calculates adiabatic temperature gradient)
- pres (calculates pressure from depth)
- T90 conversion (Convert ITS-90 temperature to IPTS-68)
- T68 conversion (Convert IPTS-68 or IPTS-48 to temperature to ITS-90)
- deg2rad constant

Given how much migration and testing would need to occur, as well as getting the script ready to run in METplus wrappers, I would say there would need to be approx 60-80 hours dedicated to this, from start to repo placement.
